src/test/java/com/spotify/docker/it/LogReaderHangTest.java

```diff
@@ -0,0 +1,142 @@
+/*-
+ * -\-\-
+ * docker-client
+ * --
+ * Copyright (C) 2016 Spotify AB
+ * --
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+*
+ *      http://www.apache.org/licenses/LICENSE-2.0
+*
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ * -/-/-
+ */
+
+
+package com.spotify.docker.it;
+
+
+import com.google.common.io.Resources;
+import com.spotify.docker.client.DefaultDockerClient;
+import com.spotify.docker.client.DockerClient;
+import com.spotify.docker.client.exceptions.DockerCertificateException;
+import com.spotify.docker.client.exceptions.DockerException;
+import com.spotify.docker.client.exceptions.ImageNotFoundException;
+import java.io.IOException;
+import java.nio.file.Paths;
+import org.joda.time.Duration;
+import org.junit.After;
+import org.junit.Before;
+import org.junit.BeforeClass;
+import org.junit.Test;
+
+/**
+ * Test for issue 552: Obtaining next log message from Docker container hangs
+ * indefinitely.
+ *
+ * @author Ivan Krizsan
+ */
+public class LogReaderHangTest {
+  /* Constant(s): */
+  public static final String LOGHANG_DOCKERIMAGE = "loghang:latest";
+
+  /* Class variable(s): */
+  protected static DockerClient dockerClient;
+
+  /* Instance variable(s): */
+  protected IvanDockerContainer ivanDockerContainer;
+
+  /**
+   * Performs setup before all the test methods.
+   *
+   * @throws DockerCertificateException If creation of Docker client failed.
+   */
+  @BeforeClass
+  public static void setUpOnceBeforeAllTests() throws DockerCertificateException {
+    dockerClient = DefaultDockerClient.fromEnv().readTimeoutMillis(10000).build();
+  }
+
+  /**
+   * Performs setup before one test method.
+   *
+   * @throws Exception If error occurred creating test Docker image.
+   */
+  @Before
+  public void setupBeforeOneTest() throws Exception {
+    createDockerLogReaderHangTestImage();
+
+    ivanDockerContainer =
+        new IvanDockerContainer(dockerClient, LOGHANG_DOCKERIMAGE);
```

Comment:
couldn't all of the logic in the IvanDockerContainer class be in this test class instead? doesn't seem worthwhile to put that logic in its own class if it is only used in one test.